sycl/doc/extensions/experimental/sycl_ext_intel_fpga_task_sequence.asciidoc

```diff
@@ -0,0 +1,498 @@
+_Object_1_Thread_ and _Object_2_Thread_. The first two `async` calls create task
+threads _Object_1_Task_1_Thread_ and _Object_1_Task_2_Thread_. Similarly the
+next two calls create _Object_2_Task_1_Thread_ and _Object_2_Task_2_Thread_.
+
+The table below provides a view of the hierarchy of task threads that will be
+generated.
+
+.Hierarchy of task threads.
+[cols="s,,,,"]
+|=====
+// row 1, cells 2 spans 4 cells hence the '4+' before '|'
+| Work Item 4+^| _WI_
+// row 2, cells after the first one span 2 cells each
+|Task Sequence Object Thread
+2+^|_Object_1_Thread_
+2+^| _Object_2_Thread_
+// row 3
+| Task Thread
+^| _Task_1_1_
+^|__Task_1_2__
+^|_Task_2_1_
+^|_Task_2_2_
+|=====
+
+At some initial stage, all task threads have weakly parallel forward progress
+guarantee. If _K_ is strengthened to have concurrent forward progress
+guarantee, then all of the object threads are also strengthened. Next, in this
+example one task thread for each task sequence is also strengthened. This is
+depicted in the table below (progress guarantee for each thread is in
+parenthesis):
+
+.Possible Progress Guarantees at some time after _WI_ is strengthened.
+[cols="s,,,,"]
+|=====
+// row 1, cells 2 spans 4 cells hence the '4+' before '|'
+| Work Item
+4+^| _WI_ (concurrently parallel)
+// row 2, cells after the first one span 2 cells each
+|Task Sequence Object Thread
+2+^|_Object_1_Thread_ (concurrent)
+2+^| _Object_2_Thread_ (concurrent)
+// row 3
+| Task Thread
+^| _Task_1_1_ (weakly parallel)
+^|__Task_1_2__ (concurrent)
+^|_Task_2_1_ (concurrent)
+^|_Task_2_2_ (weakly parallel)
+|=====
+
+The next example shows how a task thread delegates its progress
+guarantee to another task thread:
+
+Assume that we have a task sequence object _TS_ templated on the function `f`
+with `response_capacity` of 1 and `invocation_capacity` of 5. Four `async` calls
+create the following task threads: _T1_, _T2_, _T3_ and _T4_, for _TS_. _T1_ has
+concurrent forward progress guarantee after getting strengthened, while _T2_,
+_T3_ and _T4_ have weakly parallel forward progress guarantees. The task threads
+go through the following execution flow:
+
+ - _T1_ finishes executing the function `f` associated with _TS_.
+
+ - For _TS_, the output data structure _D_ can store the output of only one
+ task thread since `response_capacity` is one. _T1_ writes its output.
+
+ - Any task thread can now be picked to be strengthened to have concurrent
+ forward progress guarantee. Let's say _T2_ is picked.
+
+ - At some point _T2_ finishes executing `f`. _T1_'s results are still in the
+ output data structure.
+
+ - _T2_ cannot write its results until space is available in _D_. Hence
+ , none of the other task threads can be picked to be strengthened to the
+ stronger progress guarantee.
+
+ - `get` is invoked. _T1_'s results get removed from _D_.
+
+ - _T2_ can write its results and some other task thread can be picked to be
+ strengthened.
+
+=== Memory Order Semantics
```

Comment:
You've dropped the scoping that was present in the SPIR-V spec.  I don't think this is safe as this is a stronger wording than the SPIR-V we want to lower it to.

Reply:
Added